src/main/java/de/tum/cit/aet/artemis/atlas/dto/BatchCompetencyPreviewResponseDTO.java

```diff
@@ -0,0 +1,16 @@
+package de.tum.cit.aet.artemis.atlas.dto;
+
+import java.util.List;
+
+import org.jspecify.annotations.Nullable;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+
+/**
+ * DTO for batch competency preview response.
+ * This is the response structure when previewing multiple competencies.
+ * The client checks for "batchPreview": true to identify this response type.
+ */
+@JsonInclude(JsonInclude.Include.NON_EMPTY)
+public record BatchCompetencyPreviewResponseDTO(boolean batchPreview, int count, @Nullable List<CompetencyPreviewDTO> competencies, boolean viewOnly) {
+}
```

Comment:
I do not understand this. Why do we have a boolean `batchPreview` in `BatchCompetencyPreviewResponseDTO`?

Why do we need  a boolean to identify the response type?